services/background-worker/src/Wire/BackgroundWorker/Env.hs

```diff
@@ -127,10 +138,20 @@ instance (MonadIO m) => MonadLogger (AppT m) where
 runAppT :: Env -> AppT m a -> m a
 runAppT env app = runReaderT (unAppT app) env
 
-markAsWorking :: (MonadIO m) => Worker -> AppT m ()
-markAsWorking worker =
-  flip modifyIORef (Map.insert worker True) =<< asks statuses
-
-markAsNotWorking :: (MonadIO m) => Worker -> AppT m ()
-markAsNotWorking worker =
-  flip modifyIORef (Map.insert worker False) =<< asks statuses
+markAsWorking :: (MonadIO m, MonadMonitor m) => Worker -> AppT m ()
+markAsWorking worker = do
+  env <- ask
+  modifyIORef env.statuses $ Map.insert worker True
+  withLabel env.workerRunningGauge (Text.pack $ show worker) (flip setGauge 1)
```

Comment:
Not a big deal, but we could also consider to invent `markAs :: (...) => Bool -> Double -> AppT m ()` to reduce duplication.